apps/meteor/app/apps/server/communication/rest.js

```diff
@@ -65,6 +66,22 @@ export class AppsRestApi {
 
 		this.api.addRoute('actionButtons', ...actionButtonsHandler(this));
 
+		this.api.addRoute(
+			'incompatibleModal',
+			{ authRequired: false },
```

Comment:
What's the reason for making this not auth required?

Reply:
@graywolf336 We don't ask for the token on this modal. Should we do that? This modal doesn't execute any action, it's just read-only.

Reply:
So, even users who are not Rocket.Chat users should be able to use this endpoint?

Reply:
Do you mean the marketplace endpoint? Right now, you must provide a workspace id, but there's no need for the token. We just render the app info version and name into the modal, but for more safety, I'm considering adding this token as mandatory. What do you think? 🤔

Reply:
No sir, I was talking about the endpoint in the Rocket.Chat version. By having `{ authRequired: false }`, anyone can call that endpoint even if they are not a user of that Rocket.Chat server. Which seeing the bugs and "security" issues that come in, this will be one that is brought up and I want to make sure we have a proper response if that is the intended functionality. I think it should be `{ authRequired: true }` so that the person must be a user of that rocket.chat instance. The auth required at this endpoint is on rocket.chat's side, not Marketplace's.

The token to Marketplace side shouldn't matter, in fact most users will not have access to that setting and shouldn't have access to it because then they could grab that and act on the behalf of the admins since that token is an authorized token.

Reply:
aaah got it @graywolf336!! Thank you so much for shedding a light on that 👊 Agreed with you! The rationale behind the { authRequired: false } was a misunderstanding about that. I changed it to { authRequired: true } now ✅